test/test_string_map.cpp

```diff
@@ -98,55 +130,56 @@ TEST(test_string_map, lifecycle) {
   }
 }
 
-TEST(test_string_map, getters) {
-  auto allocator = rcutils_get_default_allocator();
+TEST_F(TestStringMap, getters_capacity_null_list) {
   rcutils_ret_t ret;
+  size_t capacity;
+  ret = rcutils_string_map_get_capacity(NULL, &capacity);
+  EXPECT_EQ(RCUTILS_RET_INVALID_ARGUMENT, ret) << rcutils_get_error_string().str;
```

Comment:
Not doing a `rcutils_reset_error` here is a bad practice in my opinion, because if this case were to be extended and another rcutils function that can fail were called after this line, it would be missing an `rcutils_reset_error` between them. Saving one call to it at the end of each case doesn't save that much, but can lead to forgetting to add it in follow up cases.

I'd prefer a helper function that gets the error and resets it, which could be the right hand side of the `<<` in the expect statement. But even that I feel is unnecessary and error prone, e.g. if the expect passes but the function still failed then the error would not be fetched and therefore not reset.

Reply:
This LGTM, I put back the `rcutils_reset_error` everywhere.